File: agri-platform/frontend/agri-admin/src/models/system/menu.ts
export interface MenuVO {
  id: number;
  menuName: string;
  parentId: number;
  orderNum: number;
  path: string;
  component: string;
  menuType: string;
  menuTypeName: string;
  visible: number;
  status: number;
  perms: string;
  icon: string;
  children?: MenuVO[];
  tenantId: string;
  createBy: string;
  createTime: string;
  remark: string;
}

export interface MenuSaveDTO {
  menuName: string;
  parentId?: number;
  orderNum?: number;
  path?: string;
  component?: string;
  menuType?: string;
  visible?: number;
  perms?: string;
  icon?: string;
  status?: number;
}

export interface MenuUpdateDTO {
  menuName?: string;
  parentId?: number;
  orderNum?: number;
  path?: string;
  component?: string;
  menuType?: string;
  visible?: number;
  perms?: string;
  icon?: string;
  status?: number;
}

File: agri-masterdata/backend/src/main/java/com/agri/masterdata/entity/Menu.java
package com.agri.masterdata.entity;

import com.agri.common.entity.BaseEntity;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("agri_md_menu")
public class Menu extends BaseEntity {

    private String menuName;

    private Long parentId;

    private Integer orderNum;

    private String path;

    private String component;

    private String menuType;

    private Integer visible;

    private Integer status;

    private String perms;

    private String icon;

    private Integer delFlag;

    private String tenantId;
}
